utils/analyticsmetrics.go

```diff
@@ -0,0 +1,204 @@
+	return ams.finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) SetFinalizeEvent(finalizeEvent *xscservices.XscAnalyticsGeneralEventFinalize) {
+	ams.finalizeEvent = finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) CreateGeneralEvent(product xscservices.ProductName, eventType xscservices.EventType) *xscservices.XscAnalyticsGeneralEvent {
+	osAndArc, err := coreutils.GetOSAndArc()
+	curOs, curArch := "", ""
+	if err != nil {
+		log.Debug(fmt.Errorf("failed to get os and arcitucture for general event request to XSC service, error: %s ", err.Error()))
+	} else {
+		splitOsAndArch := strings.Split(osAndArc, "-")
+		curOs = splitOsAndArch[0]
+		curArch = splitOsAndArch[1]
+	}
+
+	event := xscservices.XscAnalyticsGeneralEvent{
+		XscAnalyticsBasicGeneralEvent: xscservices.XscAnalyticsBasicGeneralEvent{
+			EventType:              eventType,
+			EventStatus:            xscservices.Started,
+			Product:                product,
+			JfrogUser:              ams.xscManager.Config().GetServiceDetails().GetUser(),
+			OsPlatform:             curOs,
+			OsArchitecture:         curArch,
+			AnalyzerManagerVersion: GetAnalyzerManagerVersion(),
+		},
+	}
+	return &event
+}
+
+func (ams *AnalyticsMetricsService) AddGeneralEvent(event *xscservices.XscAnalyticsGeneralEvent) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("A general event request was not sent to XSC - analytics metrics are disabled.")
+		return
+	}
+	msi, err := ams.xscManager.AddAnalyticsGeneralEvent(*event)
+	if err != nil {
+		log.Debug(fmt.Errorf("failed sending general event request to XSC service, error: %s ", err.Error()))
+		return
+	}
+	log.Debug(fmt.Sprintf("New General event added successfully. multi_scan_id %s", msi))
+	// Set event's analytics data.
+	ams.SetMsi(msi)
+	ams.SetStartTime()
+}
+
+func (ams *AnalyticsMetricsService) UpdateGeneralEvent(event *xscservices.XscAnalyticsGeneralEventFinalize) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("A general event update request was not sent to XSC - analytics metrics are disabled.")
+		return
+	}
+	if ams.msi == "" {
+		log.Debug("A general event update request was not sent to XSC - failed to add the starting event, and there is no valid multi-scan identifier provided.")
+		return
+	}
+	err := ams.xscManager.UpdateAnalyticsGeneralEvent(*event)
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed updading general event request in XSC service for multi_scan_id %s, error: %s \"", ams.GetMsi(), err.Error()))
+	}
+}
+
+func (ams *AnalyticsMetricsService) GetGeneralEvent(msi string) (*xscservices.XscAnalyticsGeneralEvent, error) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("Can't get general event from XSC - analytics metrics are disabled.")
+		return nil, nil
+	}
+	event, err := ams.xscManager.GetAnalyticsGeneralEvent(msi)
+	if err != nil {
+		log.Debug(fmt.Sprintf("failed getting general event from XSC service for multi_scan_id %s, error: %s \"", msi, err.Error()))
+	}
+	return event, err
```

Comment:
```suggestion
	return ams.xscManager.GetAnalyticsGeneralEvent(msi)
```
log can be added where this func is used, we return it...

Reply:
this function is for testing and getting the analytics from xsc - I think that its fine like this.